### PR TITLE
Add Nickname to report header and fix `RunQuery` bug

### LINKED ISSUE #1878: Feature: Present study nickname at top of report if present in `dfGroups`.
## Feature Details
For studies with nicknames display the nickname alongside the study ID at the top of the reports, like _AA-AA-000-0000 (Study Nickname)_.

## Example Code
```
gsm::reportingGroups %>%
    filter(
        .data$GroupLevel == 'Study',
        .data$Param == 'nickname'
    ) %>%
    mutate(
        StudyLabel = glue::glue('{.data$GroupID} ({.data$Value})')
    ) %>%
    pull(StudyLabel)
```

## Additional Comments
Might want to parameterize or lockdown the naming convention of the `nickname` parameter, and definitely add a logic check for `NA` values, because not all studies will have nicknames.

### LINKED ISSUE #1914: Bugfix: RunQuery returns dates improperly when query is run outside of `ApplySpec()`
## Expected Behavior

Dates should return as dates, not numeric

## Current Behavior

when not running through `ApplySpec`, RunQuery does not return dates properly, as in the first and second step of `report_kri_site.yaml` and `report_kri_country.yaml`. 

## Possible Solution

This is likely because we added `method = "name__class"` to sqldf() in `RunQuery()`. exposing the method argument to RunQuery so that different methods can be used depending on context could be a solution here

## Steps to Reproduce

1.
2.
3.
4.

### PR DESCRIPTION
**Pull request by lauramaxwell**
## Overview

Adds nickname to report header (when present) and fixes a bug that caused errors in RunQuery Dates outside of `ApplySpec()`

## Test Notes/Sample Code

Run example 3.2 in `inst/examples/3_ReportingWorkflow.R`

## Connected Issues

- Closes #1878
- Closes #1914

### CODE REVIEW

**Comment by jonthegeek** [comment]
Merged main to pick up #1923.

**Comment by jonthegeek** [comment]
Confirmed that this closes #1925!

**Review by jonthegeek — APPROVED** [review]
Looks great! I'm approving this from the context of the issue I was having (#1925, which was a duplicate of #1914, evidently), but I didn't dig into #1878.

File: R/Report_Setup.R [hunk 1]
```diff
@@ -53,6 +53,11 @@ Report_Setup <- function(dfGroups = NULL, dfMetrics = NULL, dfResults = NULL) {
     pull("GroupID") %>%
     unique()
 
+  output$StudyLabel <- ifelse(is.na(output$lStudy$nickname) || is.null(output$lStudy$nickname),
+                              output$StudyID,
+                              glue::glue('{output$lStudy$studyid} ({output$lStudy$nickname})')
```
Comment (samussiah) [inline R/Report_Setup.R:58]:
```suggestion
                              glue::glue('{output$StudyID} ({output$lStudy$nickname})')
```

Comment (samussiah) [reply R/Report_Setup.R:58]: For consistency - not sure if there's a meaningful distinction between the two.

**Review by samussiah — CHANGES_REQUESTED** [review]
Looks good @lauramaxwell!  Minor tweak for consistency, otherwise good to go!

**Comment by lauramaxwell** [comment]
> Looks good @lauramaxwell! Minor tweak for consistency, otherwise good to go!

updated this minor tweak!

**Review by samussiah — APPROVED** [review]
Perf.

**Pull request merged**